### PR TITLE
[Merged by Bors] - feat: short exact sequence of free modules

### PR DESCRIPTION
**Pull request by dagurtomas**
We prove that if the left and right term in a short exact sequence of modules are free, then the middle term is free as well, and related results. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by riccardobrasca — COMMENTED** [review]
Very nice result, thanks!!

I am not completely sure about certain names, but this not very important.

Can you add a formula for the rank?

**Comment by dagurtomas** [comment]
Thanks for the quick review! I've resolved all your comments and added the rank formula you asked for

**Review by riccardobrasca — APPROVED** [review]
Thanks! Can you also add a formula for [FiniteDimensional.finrank](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Finrank.html#FiniteDimensional.finrank) (I should have asked this before, sorry)?

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/Algebra/Category/ModuleCat/Free.lean [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+open CategoryTheory
+
+section LinearIndependent
+
+variable (hv : LinearIndependent R v)  {M : ModuleCat R}
+  {u : ι ⊕ ι' → M}  {f : N ⟶ M} {g : M ⟶ P}
+  (hw : LinearIndependent R (g ∘ u ∘ Sum.inr)) (hu : Function.Injective u)
+  (hm : Mono f) (he : Exact f g) (huv : u ∘ Sum.inl = f ∘ v)
+
+theorem disjoint_span_sum : Disjoint (Submodule.span R (Set.range (u ∘ Sum.inl)))
```
Comment (eric-wieser) [inline Mathlib/Algebra/Category/ModuleCat/Free.lean:47]: This lemma does not belong in the `ModuleCat` namesoace

**Pull request closed without merging**